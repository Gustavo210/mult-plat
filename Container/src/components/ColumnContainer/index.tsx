import React from 'react'
import styled, { css } from 'styled-components/native'

interface ColumnContainerProps {
  children: React.ReactNode
  gapSize: string
}

export default function ColumnContainer({ children, gapSize }: ColumnContainerProps) {
  return <ViewColumn $gapSize={gapSize}>{children}</ViewColumn>
}

const ViewColumn = styled.View<{
  $gapSize: string
}>`
  width: 100%;
  flex-direction: column;

  ${({ $gapSize, theme }) => css`
    gap: ${theme.gaps[$gapSize.toLowerCase()]}px;
  `}
`
