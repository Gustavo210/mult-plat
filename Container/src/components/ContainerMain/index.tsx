import React from "react";
import { View } from "react-native";
import styled, { useTheme } from "styled-components/native";

interface ContainerMainProps {
  children: React.ReactNode;
  debug?: boolean | string;
}

export function ContainerMain({ children, debug = false }: ContainerMainProps) {
  const theme = useTheme();

  return (
    <View style={{ flex: 1, position: "relative" }}>
      <PageWrapper>{children}</PageWrapper>
      {debug && (
        <ColumnContainer>
          {Array.from({
            length: Math.max(
              ...Object.values(theme.columns).filter(
                (item, index, arr) => arr.indexOf(item) === index
              )
            ),
          }).map((_, index) => (
            <VirtualizedColumn key={index} debug={debug} />
          ))}
        </ColumnContainer>
      )}
    </View>
  );
}

const PageWrapper = styled.View`
  width: 100%;
  flex-direction: column;
  padding: 0px 4px 12px 4px;
`;

const ColumnContainer = styled.View`
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  gap: 16px;
  flex-direction: row;
  padding: 0px 4px 12px 4px;
  z-index: -1;
`;

const VirtualizedColumn = styled.View.attrs({
  testID: "virtual-column",
})<{ debug: boolean | string }>`
  flex: 1;
  background-color: ${({ debug }) =>
    typeof debug === "string" ? `${debug}` : "#c8348848"};
  opacity: 0.3;
  border: 1px dashed
    ${({ debug }) => (typeof debug === "string" ? debug : "#c83488")};
`;
