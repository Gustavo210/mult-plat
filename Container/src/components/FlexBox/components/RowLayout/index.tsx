import React from "react";
import { LayoutChangeEvent } from "react-native";
import styled, { css } from "styled-components/native";

import { FlexBoxProps } from "../..";

interface RowLayoutProps {
  children: React.ReactNode;
  gapSize: string;
  align?: FlexBoxProps["align"];
  onLayout(event: LayoutChangeEvent): void;
}

export default function RowLayout({
  children,
  gapSize,
  align,
  onLayout,
}: RowLayoutProps) {
  return (
    <RowContainer $gapSize={gapSize} $align={align} onLayout={onLayout}>
      {children}
    </RowContainer>
  );
}

const RowContainer = styled.View<{
  $gapSize: string;
  $align?: FlexBoxProps["align"];
}>`
  flex-direction: row;

  ${({ $gapSize, theme }) => css`
    gap: ${theme.gaps[$gapSize.toLowerCase()]}px;
  `}

  ${({ $align }) => {
    switch ($align) {
      case "CENTER":
        return css`
          justify-content: center;
        `;
      case "RIGHT":
        return css`
          justify-content: flex-end;
        `;
      case "SPACE_BETWEEN":
        return css`
          justify-content: space-between;
        `;
      default:
        return css`
          justify-content: flex-start;
        `;
    }
  }}
`;
