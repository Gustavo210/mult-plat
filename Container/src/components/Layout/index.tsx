import React from "react";
import { View } from "react-native";
import styled, { useTheme } from "styled-components/native";

interface LayoutProps {
  children: React.ReactNode;
  debug?: boolean | string;
}

export function Layout(props: LayoutProps): JSX.Element {
  const theme = useTheme();

  return (
    <View style={{ flex: 1, position: "relative" }}>
      <PageWrapper>{props.children}</PageWrapper>
      {props.debug && (
        <ColumnContainer>
          {Array.from({
            length: Math.max(
              ...Object.values(theme.columns).filter(
                (item, index, arr) => arr.indexOf(item) === index
              )
            ),
          }).map((_, index) => (
            <VirtualizedColumn key={index} />
          ))}
        </ColumnContainer>
      )}
    </View>
  );
}

const PageWrapper = styled.View`
  width: 100%;
  flex-direction: column;
  padding: 0px 4px 12px 4px;
`;

const ColumnContainer = styled.View`
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  gap: 16px;
  flex-direction: row;
  padding: 0px 4px 12px 4px;
  z-index: -1;
`;

const VirtualizedColumn = styled.View.attrs({
  testID: "virtual-column",
})`
  flex: 1;
  background-color: #c8348848;
  opacity: 0.3;
  border: 1px dashed #c83488;
`;
