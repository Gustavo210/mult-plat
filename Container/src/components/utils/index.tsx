import { css } from "styled-components/native";

export function calculateAvailableSpace(
  containerWidth: number,
  columns: number,
  gap: number
) {
  if (columns <= 0) return 0;
  const singleColWidth = (containerWidth - gap * 3) / 4;
  return singleColWidth * columns + gap * (columns - 1);
}

export function getAlignmentStyles($align, $noFlex) {
  // Essa função será removida, está aqui somente para referência
  if (!$align) return "";

  if ($noFlex) {
    const flexValues = {
      LEFT: "flex-start",
      CENTER: "center",
      RIGHT: "flex-end",
      SPACE_BETWEEN: "space-between",
    };
    const value = flexValues[$align];
    return value
      ? css`
          justify-content: ${value};
        `
      : "";
  }

  const marginStyles = {
    LEFT: css`
      margin-right: auto;
    `,
    CENTER: css`
      margin-left: auto;
      margin-right: auto;
    `,
    RIGHT: css`
      margin-left: auto;
    `,
  };
  return marginStyles[$align] || "";
}

export function AlignFlexItems($align: string) {
  // Em teoria, isso deveria funcionar nos itens flex. mas funciona somente nos que recebem a prop noFlex

  if (!$align) return "";

  const flexValues = {
    LEFT: "flex-start",
    CENTER: "center",
    RIGHT: "flex-end",
    SPACE_BETWEEN: "space-between",
  };

  const value = flexValues[$align];
  return value
    ? css`
        justify-content: ${value};
      `
    : "";
}

export function AlignNoFlexItems($align: string) {
  // Em teoria, isso deveria funcionar nos itens não flex. mas funciona somente nos que não recebem a prop noFlex

  if (!$align) return "";

  const marginStyles = {
    LEFT: css`
      margin-right: auto;
    `,
    CENTER: css`
      margin-left: auto;
      margin-right: auto;
    `,
    RIGHT: css`
      margin-left: auto;
    `,
  };

  return marginStyles[$align] || "";
}
