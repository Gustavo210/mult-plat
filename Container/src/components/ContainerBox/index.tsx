import React, { useState } from "react";
import { LayoutChangeEvent } from "react-native";
import styled, { DefaultTheme, css, useTheme } from "styled-components/native";

import { AlignOptions } from "../..";
import ColumnContainer from "../ColumnContainer";
import RowContainer from "../RowContainer";
import {
  AlignFlexItems,
  AlignNoFlexItems,
  calculateAvailableSpace,
  getAlignmentStyles,
} from "../utils";

export interface ContainerBoxProps {
  children: React.ReactNode;
  direction?: "ROW" | "COLUMN";
  sizeKey?: Uppercase<keyof DefaultTheme["columns"] & string>;
  gapSize?: Uppercase<keyof DefaultTheme["gaps"] & string>;
  align?: AlignOptions;
  noFlex?: boolean;
  debug?: boolean | string;
}

export function ContainerBox({
  children,
  direction = "ROW",
  gapSize,
  align,
  noFlex = false,
  debug = false,
}: ContainerBoxProps) {
  const theme = useTheme();
  const [containerWidth, setContainerWidth] = useState(0);
  const isRow = direction === "ROW";
  const gapSizeValue = gapSize || (isRow ? "MD" : "NONE");

  const gap = theme.gaps[gapSizeValue.toLowerCase()];

  function onLayoutContainer(event: LayoutChangeEvent) {
    setContainerWidth(event.nativeEvent.layout.width);
  }

  if (direction === "ROW") {
    return (
      <RowContainer
        gapSize={gapSizeValue}
        align={align}
        onLayout={onLayoutContainer}
      >
        {React.Children.map(children, (child) => {
          if (!React.isValidElement(child)) return null;

          const columnKey = child.props.size?.toLowerCase?.();
          const columns = columnKey ? theme.columns[columnKey] || 1 : null;
          let availableSpace;

          if (columns && containerWidth) {
            availableSpace = calculateAvailableSpace(
              containerWidth,
              columns,
              gap
            );
          }

          return (
            <FlexItem
              $direction="ROW"
              $availableSpace={availableSpace}
              $noFlex={child.props.noFlex ?? noFlex}
              $debug={debug}
              $align={child.props.align}
            >
              {child}
            </FlexItem>
          );
        })}
      </RowContainer>
    );
  }

  return <ColumnContainer gapSize={gapSizeValue}>{children}</ColumnContainer>;
}

const FlexItem = styled.View.attrs({ testID: "flex-item-wrapper" })<{
  $align?: AlignOptions;
  $direction?: "ROW" | "COLUMN";
  $availableSpace?: number;
  $noFlex?: boolean;
  $debug?: boolean | string;
}>`
  ${({ $debug, theme }) =>
    $debug &&
    css`
      border-width: 1px;
      border-style: solid;
      border-color: ${typeof $debug === "string"
        ? $debug
        : theme.colors.container.visibleArea};
    `}

  ${({ $availableSpace, $noFlex }) => {
    if ($availableSpace) {
      return css`
        width: ${$availableSpace}px;
      `;
    }

    if ($noFlex) {
      return css`
        width: auto;
      `;
    }

    return css`
      flex: 1;
    `;
  }}

    ${({ $align, $noFlex }) =>
    $noFlex ? AlignFlexItems($align) : AlignNoFlexItems($align)};
  /* ${({ $align, $noFlex }) => getAlignmentStyles($align, $noFlex)} */
`;
