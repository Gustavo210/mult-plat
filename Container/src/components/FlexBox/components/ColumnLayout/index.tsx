import React from "react";
import { LayoutChangeEvent } from "react-native";
import styled, { css } from "styled-components/native";

import { FlexBoxProps } from "../..";
import { getAlignmentStyles } from "../../../utils";

interface RowLayoutProps {
  children: React.ReactNode;
  gapSize: string;
  align?: FlexBoxProps["align"];
  onLayout(event: LayoutChangeEvent): void;
}

export default function RowLayout({
  children,
  gapSize,
  align,
  onLayout,
}: RowLayoutProps) {
  return (
    <RowContainer $gapSize={gapSize} $align={align} onLayout={onLayout}>
      {children}
    </RowContainer>
  );
}

const RowContainer = styled.View<{
  $gapSize: string;
  $align?: FlexBoxProps["align"];
}>`
  flex-direction: row;

  ${({ $gapSize, theme }) => css`
    gap: ${theme.gaps[$gapSize.toLowerCase()]}px;
  `}

  ${({ $align }) => getAlignmentStyles($align, true)}
`;
