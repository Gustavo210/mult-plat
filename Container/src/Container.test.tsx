import '@testing-library/jest-native/extend-expect'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import React from 'react'
import { Text } from 'react-native'

import { Container } from '.'
import { ContainerBox } from './components/ContainerBox'
import { ContainerMain } from './components/ContainerMain'
import { calculateAvailableSpace, getAlignmentStyles } from './components/utils'

describe('Container component tests', () => {
  it('should match snapshot with default variant', () => {
    const { toJSON } = render(
      global.app(
        <Container>
          <Text>Default</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should match snapshot with variant="MAIN"', () => {
    const { toJSON } = render(
      global.app(
        <Container variant="MAIN">
          <Text>Main Variant</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should match snapshot with variant="BOX" and direction="COLUMN"', () => {
    const { toJSON } = render(
      global.app(
        <Container variant="BOX" direction="COLUMN">
          <Text>Box Column</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should match snapshot with direction="ROW"', () => {
    const { toJSON } = render(
      global.app(
        <Container direction="ROW">
          <Text>Row</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should align content to center visually with align="CENTER"', () => {
    const { toJSON } = render(
      global.app(
        <Container>
          <Container size="SM" align="CENTER">
            <Text>Centralizado</Text>
          </Container>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should align content to the right visually with align="RIGHT"', () => {
    const { toJSON } = render(
      global.app(
        <Container>
          <Container size="SM" align="RIGHT">
            <Text>Right Aligned</Text>
          </Container>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should apply flex-basis when sizeKey is provided', () => {
    const { toJSON } = render(
      global.app(
        <Container>
          <Container size="SM">
            <Text>Sized</Text>
          </Container>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should not affect alignment from parent align prop', () => {
    const { getAllByTestId } = render(
      global.app(
        <Container align="CENTER">
          <Container size="SM">
            <Text>Inner</Text>
          </Container>
        </Container>
      )
    )
    const wrappers = getAllByTestId('flex-item-wrapper')
    const childWrapper = wrappers[1]
    expect(childWrapper).toBeTruthy()
    expect(childWrapper).not.toHaveStyle({ justifyContent: 'center' })
  })

  it('should render multiple children correctly', () => {
    const { getByText } = render(
      global.app(
        <Container>
          <Text>One</Text>
          <Text>Two</Text>
        </Container>
      )
    )
    expect(getByText('One')).toBeTruthy()
    expect(getByText('Two')).toBeTruthy()
  })

  it('should distribute space equally between children by default', () => {
    const { getAllByTestId } = render(
      global.app(
        <Container>
          <Container>
            <Text>Child 1</Text>
          </Container>
          <Container>
            <Text>Child 2</Text>
          </Container>
        </Container>
      )
    )
    const wrappers = getAllByTestId('flex-item-wrapper')
    const [child1, child2] = [wrappers[1], wrappers[2]]
    expect(child1.props.style).toMatchObject({ flexGrow: 1 })
    expect(child2.props.style).toMatchObject({ flexGrow: 1 })
  })

  it('should render plain string child without crashing', () => {
    const { getByText } = render(
      global.app(
        <Container>
          <Text>One</Text>
        </Container>
      )
    )
    expect(getByText('One')).toBeTruthy()
  })

  it('should apply default gapSize (MD) when gap is not provided', () => {
    const { toJSON } = render(
      global.app(
        <Container variant="BOX" direction="ROW">
          <Text>Child 1</Text>
          <Text>Child 2</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should apply specific gap when gap="LG"', () => {
    const { toJSON } = render(
      global.app(
        <Container variant="BOX" direction="ROW" gap="LG">
          <Text>With LG Gap</Text>
          <Text>Another</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should fallback to 0px gap when gap is invalid', () => {
    const { toJSON } = render(
      global.app(
        <Container variant="BOX" direction="ROW" gap={'INVALID' as never}>
          <Text>Invalid Gap</Text>
          <Text>Another</Text>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should skip invalid children', () => {
    const { toJSON } = render(
      global.app(
        <Container>
          {null}
          {false}
          {'just text'}
          <Container>
            <Text>Valid</Text>
          </Container>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should fallback to 1 column if theme.columns[size] is undefined', () => {
    const { toJSON } = render(
      global.app(
        <Container direction="ROW">
          <Container size={'INVALID' as never}>
            <Text>Fallback Columns</Text>
          </Container>
        </Container>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should apply debug border styles when debug is true', () => {
    const { getAllByTestId } = render(
      global.app(
        <Container direction="ROW" debug>
          <Container size="SM">
            <Text>Debug Border</Text>
          </Container>
        </Container>
      )
    )
    const wrappers = getAllByTestId('flex-item-wrapper')
    const debugWrapper = wrappers[0]
    expect(debugWrapper).toHaveStyle({
      borderWidth: 1,
      borderColor: 'red',
      borderStyle: 'solid'
    })
  })
})

describe('FlexBox component tests', () => {
  it('should grow flexibly when sizeKey is not provided', () => {
    const { getAllByTestId } = render(
      global.app(
        <Container>
          <Container>
            <Text>Flexible</Text>
          </Container>
        </Container>
      )
    )
    const wrappers = getAllByTestId('flex-item-wrapper')
    const childWrapper = wrappers[1]
    expect(childWrapper).toBeTruthy()
    expect(childWrapper.props.style).toMatchObject({ flexGrow: 1 })
  })

  it('should trigger onLayout and calculate available space', async () => {
    const { getAllByTestId, toJSON } = render(
      global.app(
        <Container>
          <Container size="SM" align="CENTER">
            <Text>Layout Event</Text>
          </Container>
        </Container>
      )
    )

    const wrapper = getAllByTestId('flex-item-wrapper')[0]
    const container = wrapper.parent

    fireEvent(container, 'layout', {
      nativeEvent: { layout: { width: 300 } }
    })

    await waitFor(() => {
      expect(wrapper).toBeTruthy()
    })

    expect(toJSON()).toMatchSnapshot()
  })

  it('should trigger availableSpace and calculate childAvailableSpace', async () => {
    const { getAllByTestId, toJSON } = render(
      global.app(
        <Container direction="ROW">
          <Container size="SM" align="LEFT">
            <Text>Child</Text>
          </Container>
        </Container>
      )
    )

    const wrapper = getAllByTestId('flex-item-wrapper')[0]
    const container = wrapper.parent

    fireEvent(container, 'layout', {
      nativeEvent: { layout: { width: 320 } }
    })

    await waitFor(() => {
      expect(wrapper).toBeTruthy()
    })

    expect(toJSON()).toMatchSnapshot()
  })

  it('should apply width: auto when noFlex is true', () => {
    const { toJSON } = render(
      global.app(
        <Container direction="ROW" noFlex={true}>
          <Container size="MD">
            <Text>NoFlex Test</Text>
          </Container>
        </Container>
      )
    )
    const tree = toJSON()
    const children = tree?.children || []
    const wrappers = children.filter(child => child?.props?.testID === 'flex-item-wrapper')

    expect(wrappers[0]).toHaveStyle({ width: 'auto' })
  })

  it('should use default props when none are provided', () => {
    const { toJSON } = render(
      global.app(
        <ContainerBox>
          <Text>Default Props</Text>
        </ContainerBox>
      )
    )
    expect(toJSON()).toMatchSnapshot()
  })
})

describe('Layout component tests', () => {
  it('should render virtualized columns when debug is true', () => {
    const { getAllByTestId } = render(
      global.app(
        <ContainerMain debug>
          <Text>Test</Text>
        </ContainerMain>
      )
    )
    const virtualColumns = getAllByTestId('virtual-column')
    expect(virtualColumns.length).toBeGreaterThan(0)
  })
})

describe('Utility functions tests', () => {
  it('should return empty string if no $align is provided', () => {
    const result = getAlignmentStyles(undefined, false)
    expect(result).toBe('')
  })

  it('should return justify-content when $noFlex is true', () => {
    expect(getAlignmentStyles('LEFT', true).join('')).toContain('justify-content: flex-start')
    expect(getAlignmentStyles('CENTER', true).join('')).toContain('justify-content: center')
    expect(getAlignmentStyles('RIGHT', true).join('')).toContain('justify-content: flex-end')
    expect(getAlignmentStyles('SPACE_BETWEEN', true).join('')).toContain('justify-content: space-between')
  })

  it('should return margin styles when $noFlex is false', () => {
    expect(getAlignmentStyles('LEFT', false).join('')).toContain('margin-right: auto')
    expect(getAlignmentStyles('CENTER', false).join('').replace(/\s/g, '')).toContain(
      'margin-left:auto;margin-right:auto'
    )
    expect(getAlignmentStyles('RIGHT', false).join('')).toContain('margin-left: auto')
    expect(getAlignmentStyles('SPACE_BETWEEN', false)).toBe('')
  })

  it('should return empty string when $noFlex is true and $align is invalid', () => {
    const result = getAlignmentStyles('INVALID', true)
    expect(result).toBe('')
  })

  it('should calculate space for two columns', () => {
    const containerWidth = 400
    const qtdColumns = 2
    const qtdGaps = 3
    const gap = 16

    const availableSpace = calculateAvailableSpace(containerWidth, qtdColumns, gap)

    const width = 400
    const columns = 2
    const totalColumns = 4
    const gapValue = 16
    const totalGaps = gapValue * qtdGaps

    const expected = ((width - totalGaps) / totalColumns) * columns + gapValue * (columns - 1)

    expect(availableSpace).toBeCloseTo(expected)
  })

  it('should return 0 when columns is 0', () => {
    const containerWidth = 400
    const qtdColumns = 0
    const qtdGaps = 3
    const gap = 16

    const availableSpace = calculateAvailableSpace(containerWidth, qtdColumns, gap)

    const width = 400
    const columns = 0
    const totalColumns = 4
    const gapValue = 16
    const totalGaps = gapValue * qtdGaps

    // rawCalculation = ((400 - 16 * 3) / 4) * 0 + 16 * (0 - 1)
    const rawCalculation = ((width - totalGaps) / totalColumns) * columns + gapValue * (columns - 1)

    // expect 0 because rawCalculation its < 0, so the function returns 0
    expect(rawCalculation).toBeLessThan(0)
    expect(availableSpace).toBe(0)
  })

  it('should work with gap = 0', () => {
    const containerWidth = 400
    const qtdColumns = 3
    const qtdGaps = 3
    const gap = 0

    const availableSpace = calculateAvailableSpace(containerWidth, qtdColumns, gap)

    const width = 400
    const columns = 3
    const totalColumns = 4
    const gapValue = 0
    const totalGaps = gapValue * qtdGaps

    const expected = ((width - totalGaps) / totalColumns) * columns + gapValue * (columns - 1)

    expect(availableSpace).toBe(expected)
  })
})
