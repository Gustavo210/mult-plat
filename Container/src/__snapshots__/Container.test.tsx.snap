// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component tests should align content to center visually with align="CENTER" 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginLeft": "auto",
        "marginRight": "auto",
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Centralizado
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Container component tests should align content to the right visually with align="RIGHT" 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginLeft": "auto",
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Right Aligned
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Container component tests should apply default gapSize (MD) when gap is not provided 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Child 1
    </Text>
  </View>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Child 2
    </Text>
  </View>
</View>
`;

exports[`Container component tests should apply flex-basis when sizeKey is provided 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Sized
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Container component tests should apply specific gap when gap="LG" 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 20,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      With LG Gap
    </Text>
  </View>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Another
    </Text>
  </View>
</View>
`;

exports[`Container component tests should fallback to 0px gap when gap is invalid 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": "px",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Invalid Gap
    </Text>
  </View>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Another
    </Text>
  </View>
</View>
`;

exports[`Container component tests should fallback to 1 column if theme.columns[size] is undefined 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Fallback Columns
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Container component tests should match snapshot with default variant 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Default
    </Text>
  </View>
</View>
`;

exports[`Container component tests should match snapshot with direction="ROW" 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Row
    </Text>
  </View>
</View>
`;

exports[`Container component tests should match snapshot with variant="BOX" and direction="COLUMN" 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "gap": 0,
      "width": "100%",
    }
  }
>
  <Text>
    Box Column
  </Text>
</View>
`;

exports[`Container component tests should match snapshot with variant="MAIN" 1`] = `
<View
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "paddingBottom": 12,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <Text>
      Main Variant
    </Text>
  </View>
</View>
`;

exports[`Container component tests should skip invalid children 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Valid
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`FlexBox component tests should trigger availableSpace and calculate childAvailableSpace 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "marginRight": "auto",
        "width": 152,
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Child
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`FlexBox component tests should trigger onLayout and calculate available space 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": 142,
      }
    }
    testID="flex-item-wrapper"
  >
    <View
      onLayout={[Function]}
      style={
        {
          "flexDirection": "row",
          "gap": 16,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="flex-item-wrapper"
      >
        <Text>
          Layout Event
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`FlexBox component tests should use default props when none are provided 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexDirection": "row",
      "gap": 16,
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="flex-item-wrapper"
  >
    <Text>
      Default Props
    </Text>
  </View>
</View>
`;
